scipy/interpolate/tests/test_bsplines.py

```diff
@@ -942,6 +942,29 @@ def test_linear(self):
         b = make_interp_spline(self.xx, self.yy, k=1, axis=-1)
         assert_allclose(b(self.xx), self.yy, atol=1e-14, rtol=1e-14)
 
+    @pytest.mark.parametrize('k', [0, 1, 2, 3])
+    def test_incompatible_x_y(self, k):
+        x = [0, 1, 2, 3, 4, 5]
+        y = [0, 1, 2, 3, 4, 5, 6, 7]
+        with assert_raises(ValueError):
```

Comment:
Please match the raise statements. It will help see the logic is correct.

Reply:
OK, done. 

FTR, I do not believe having a `match=...`  clause is always a requirement. The readability improvement is often marginal at best, and the error messages themselves are not a part of the API. IOW, rephrasing the error message should not invalidate tests IMO.

Reply:
> ... and the error messages themselves are not a part of the API.

I've asked about this before, and your statement reflects my understanding too:  the precise wording of the message in an exception is not guaranteed to remain unchanged.  For our tests, that is not a problem.  If a message is changed, and an existing test relies on the old message, we just fix the test to match the new message.

> I do not believe having a `match=...` clause is always a requirement. 

I think you are correct, we have never made this a *requirement*.  But personally, I think it is valuable enough (especially with a generic exception such as `ValueError`) that is worthwhile considering it to be a "best practice", so we don't pass a test that is actually raising an exception for an unexpected reason.

Reply:
Note the developer guidelines state that use `match` is the "preferred style": [Testing that expected exceptions are raised](https://docs.scipy.org/doc/scipy/dev/missing-bits.html#testing-that-expected-exceptions-are-raised)

Reply:
Sure but in this case as we have a few code paths to raise, so we want to make sure the correct path is used. E.g. ou could have another path raising and think you are raising using the last raise while you are not.

Reply:
@tupui, if you are responding to my comments, then we are in agreement.

Reply:
I don't disagree in general, as long as it's indeed a *recommendation*, which PR authors and reviewers can apply or not apply depending on their judgement of a particular patch. In my experience, this does sometimes help finding edge cases, while in others it only leads to unnecessary  churn with little benefit. Thus, a recommendation not a requirement.

All that said, the comment is addressed, and after two CI flushes and a helpful suggestion from Jake (thanks!), this CI is green modulo what I believe is an unrelated issue with some doc page.